src/nn/embedding.rs

```diff
@@ -129,6 +135,134 @@ impl<const VOCAB: usize, const DIM: usize, D1: Device<f32>, D2: Device<f32>> ToD
     }
 }
 
+/// A learned positional embedding
+/// Initializes [Self::weight] from a Uniform distribution
+/// between [-1 / sqrt(I), 1 / sqrt(I)].
+///
+/// # Generics
+/// - `MAX_LEN` The maximum length of the input sequences. This determines how many positional embeddings to train.
+/// - `DIM` The "output" size of vectors & matrices which are the vectors being selected.
+///
+/// # Examples
+/// `LearnedPositionalEmbedding<5, 2>` can act on vectors with SEQ x DIM elements (pre-embedded),
+/// resulting in output vectors of the same size with positional embeddings added.
+///
+/// ```rust
+///
+/// # use dfdx::prelude::*;
+/// # let dev: Cpu = Default::default();
+/// type Model = (Embedding<7, 2>, LearnedPositionalEmbedding<20, 2>);
+/// let mut model = Model::build_on_device(&dev);
+/// // single sequence of ids
+/// let inputs: Tensor<Rank1<5>, usize, _> = dev.zeros();
+/// let _: Tensor<(Const<5>, Const<2>,), f32, _> = model.forward(inputs);
+/// // batched sequence of ids
+/// let inputs: Tensor<Rank2<10, 5>, usize, _> = dev.zeros();
+/// let _: Tensor<(Const<10>, Const<5>, Const<2>), f32, _> = model.forward(inputs);
+/// ```
+#[derive(Debug, Clone)]
+pub struct LearnedPositionalEmbedding<
```

Comment:
Let's move this to it's own file, also thoughts on just calling it `PositionalEmbedding`?

Reply:
I would advise actually against adding such a layer. 

It's only extremely thin addition on top of `Embedding`. It's not very flexible (for instance roberta position ids start at 2 for some reason https://github.com/huggingface/transformers/issues/10736 which this has no way to represent). 

Seems like this belongs in user code imho.